sourcemash/frontend/static/js/collections/categories.js

```diff
@@ -1,10 +1,34 @@
 Sourcemash.Collections.Categories = Backbone.Collection.extend({
 	  model: Sourcemash.Models.Category,
+
     url: '/api/categories',
+
+    initialize : function(){
+      _.bindAll(this, 'fetchResults');
+    },
+
 	  parse: function(response) {
 	  	return response.categories;
 	  },
+
 	  comparator: function(category) {
-	  	return [!category.get('unread'), category.get('name')]
-	  }
+	  	return [!category.get('unread'), category.get('name')];
+	  },
+
+    startPolling: function(item_url) {
+      $.get("/api/categorizer", {'url': item_url}, this.fetchResults);
+    },
+
+    fetchResults: function(data) {
+      var callbackFn = _.bind(function(data, status, headers) {
+        if (headers.status === 200){
+          this.add(data.categories);
+          clearInterval(poller);
+        }
+      }, this);
+
+      var poller = setInterval(function() {
+        $.get('/api/categorizer/results', {'job_id': data.job_id}, callbackFn)
+      }, 3000);
```

Comment:
The setInterval function starts the repeated ajax calls. The callback function is called after each one, so when the status is finally successful, we stop the calls with clearInterval and add the categories to our collection.